Backend-Challenges.md

```diff
@@ -65,32 +80,37 @@ In `Challenge_6.t.sol`, we want to get the token URI from the contract given a c
 
 **HINT**: Solidity will not accept string and uint concatenation. You can use the provided Strings library to turn uint into a string.
 
-### Solidity libraries and openzeppelin
+### TokenURI
 
-In Solidity, you can import libraries that will make life easier. OpenZeppelin is a company that provides free standardized contract libraries to use, such as ERC721. These contracts are firmly audited, meaning they are generally secure to implement in your contracts.
+If you are familiar with [Opensea](https://opensea.io/), you might be wondering: how on earth do they get the NFT images and metadata? Well, often times these images are stored on chain, on IPFS, or even in the cloud. As on-chain storage is expensive, images are more often than not stored off-chain.
 
-ERC721 is a standard for NFT smart contracts. There are others, such as ERC1155 by Enjin and ERC721A by Azuki, each with its own set of functionalities. For the workshop, ERC721 NFTs will suffice.
+Your smart contract provides a way to tell Opensea which image belongs to which NFT and where the files are stored through a function called `tokenURI()`. Opensea simply looks for this function in your smart contract and parses the return value.
 
-- https://docs.openzeppelin.com/contracts/3.x/
-- https://docs.openzeppelin.com/contracts/3.x/api/token/erc721
+## Challenge 7 - Minting
 
-### Constructors
+The goal of this challenge is to create a minting function in `Challenge_7.sol`, so that the tests in `Challenge_7.t.sol` pass. The mint function should:
 
-A constructor is a special function that is used to initialize state variables. An ERC721 contract, for example, requires two constructor parameters called "Name" and "Symbol," i.e., "Ethereum" and "ETH." Also, in this case, we set our max allowed supply of NFTs via the constructor.
+- Accept a uint parameter for the mint amount
+- Increase the `totalSupply` by the mint amount
+- Mint the NFTs to `msg.sender` by using ERC721's `_mint()` function.
 
-When you deploy a smart contract, you will pass these arguments to the constructor.
+Some clarification on the new things you will encounter in the test file:
 
-### TokenURI
+- `testMintEOA()` mints a token from an EOA (Externally Owned Account.) That means it simulates a human is calling the mint function, not a contract, but more on this later.
```

Comment:
Not perse right, it just means external? Could also be an arbitrage bot, or any other kind of bot. (Indirectly owned by a human but not directly). So i'd opt for: That means it simulates an off-chain user (human or program) is calling the mint function, not a contract or anything operating directly onchain, but more on this later

Reply:
Good one